airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -610,7 +611,18 @@ def delete_objects(self, bucket, keys):
         if isinstance(keys, str):
             keys = [keys]
 
-        delete_dict = {"Objects": [{"Key": k} for k in keys]}
-        response = self.get_conn().delete_objects(Bucket=bucket, Delete=delete_dict)
-
-        return response
+        s3 = self.get_conn()
+
+        # We can only send a maximum of 1000 keys per request.
+        # For details see:
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.delete_objects
+        for chunk in chunks(keys, chunk_size=1000):
+            response = s3.delete_objects(
+                Bucket=bucket,
+                Delete={"Objects": [{"Key": k} for k in chunk]}
+            )
+            deleted_keys = [x['Key'] for x in response.get("Deleted", [])]
+            self.log.info("Deleted: %s", deleted_keys)
+            if "Errors" in response:
+                errors_keys = [x['Key'] for x in response.get("Errors", [])]
+                raise AirflowException("Errors when deleting: {}".format(errors_keys))
```

Comment:
LGTM, but please mention in the Updating.md that you removed the return statement so that people are aware of this when updating :)

Some users might have implemented their own operator based on this `delete_objects` method.

Reply:
Done